service/history/replication/dlq_handler.go

```diff
@@ -257,6 +258,7 @@ func (r *dlqHandlerImpl) readMessagesWithAckLevel(
 		GetHistoryTasksRequest: persistence.GetHistoryTasksRequest{
 			ShardID:             r.shard.GetShardID(),
 			TaskCategory:        tasks.CategoryReplication,
+			ReaderID:            common.DefaultQueueReaderID,
```

Comment:
This field probably doesn't make sense for replication DLQ. 

I may later refactor the code so that GetReplicationDLQTaskRequest no longer embeds GetHistoryTasksRequest. May also do it for those dlq requests, haven't decided yet. We can sync later.
Not a problem right now cause persistence impl won't look at this field.

cc @yux0